src/pool/fifo_wait.c

```diff
@@ -246,6 +249,30 @@ static int pool_remove(ABT_pool pool, ABT_unit unit)
     return ABT_SUCCESS;
 }
 
+static int pool_print_all(ABT_pool pool, void *arg,
+                          void (*print_fn)(void *, ABT_unit)) {
+    ABT_pool_access access;
+    ABTI_pool *p_pool = ABTI_pool_get_ptr(pool);
+    void *data = ABTI_pool_get_data(p_pool);
+    data_t *p_data = pool_get_data_ptr(data);
+
+    ABT_pool_get_access(pool, &access);
+    pthread_mutex_lock(&p_data->mutex);
```

Comment:
Should you not better use ABTI_spinlock_acquire() here ??

Reply:
1. Do we need a lock here?
This pool might be used by other execution streams which are still running. If we remove it, the print function might try to print an unit that has been already popped, executed, and freed.
2. Why pthread_mutex_lock, not ABTI_spinlock_acquire?
FIFO_WAIT pool uses pthread_mutex_lock for timed_wait. See #54 for details.
3. Doesn't it cause a deadlock?
If the execution stream taking the lock hangs, yes.

Reply:
Well the changes you have pointed for #54 should have landed recently because I don't have them in my cloned argobot git tree ! Only pthread_mutex_lock() user I have is for ABTI_THREAD_HTABLE_LOCK() when defined(USE_PTHREAD_MUTEX)...

And I was already convinced that taking the lock was required ;-)